genson/src/main/java/com/owlike/genson/stream/JsonWriter.java

```diff
@@ -241,10 +241,20 @@ public JsonWriter writeValue(int value) {
   }
 
   public JsonWriter writeValue(final double value) {
-    checkValidJsonDouble(value);
-    clearMetadata();
-    beforeValue();
-    writeToBuffer(Double.toString(value), 0);
+      boolean allowExtendedNumberTypes=true;  //this should come from gensonbuilder?
```

Comment:
To add this option you will need to:
 - Add a property serializeSpecialFloatingPointNumbers to JsonWriter, it should be passed from the constructor.
 - Add to GensonBuilder a new method called serializeSpecialFloatingPointNumbers, which would be used to enable/disable this behaviour. By default it should be disabled and throw an exception on nan and inifinity.
 - Pass it to the constructor of Genson. In Genson class you have some createWriter methods, which instanciate JsonWriter with its config. Pass the option there.